### PR TITLE
Arsen/feat bacalhau bundle

### PR DESCRIPTION
**Pull request by arsen3d**
### Summary
This version implements --standalone flag.
When the flag is passed in, bacalhau and ipfs are decompressed and executed automatically, without having to manually install or start them. If bacalhau or ipfs crashes, it is automatically restarted.

To test just pass --standalone flag.
```./stack resource-provider --standalone
```
use 
```
lsof -i :4222 
```
and
```
lsof -i :5001
```
to find pid.
then use  kill command to end process.
It should automatically restart if resource-provider is running.

### CODE REVIEW

File: CHANGELOG.md
Comment (bgins) [inline CHANGELOG.md:1]: We should remove this. Only the release bot should make changes to the changelog.

**Review by bgins — CHANGES_REQUESTED** [review]
Let's discuss this with the team before moving forward.

**Comment by hunjixin** [comment]
I don't think its a good ideat to add binary in the codebase this PR has increased nearly 100m capacity.

**Comment by bgins** [comment]
> I don't think its a good ideat to add binary in the codebase this PR has increased nearly 100m capacity.

Yes, agreed. As discussed internally, it comes out to a ~3x size increase to the binary.

@arsen3d if you are still interested in pursuing this idea, could you open an issue (internal or external) outlining the benefits and tradeoffs? We should have a long-form discussion to properly consider it.

**Comment by bgins** [comment]
Closing this PR for now. We can revisit in the future if we decide to add this type of bundling.

**Pull request closed without merging**